pallets/staking/src/pallet/impls.rs

```diff
@@ -973,7 +980,11 @@ impl<T: Config> Pallet<T> {
             };
 
             if Validators::<T>::contains_key(&target) {
-                all_targets.push(target);
+                if Self::is_validator_compliant(&target)
+                    && Self::is_validator_active_balance_valid(&target)
+                {
+                    all_targets.push(target);
+                }
```

Comment:
We will also need a counter here to use instead of `all_targets.len()` in the weight calc below.